src/poetry/utils/env/python/installer.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import annotations
+
+import dataclasses
+
+from subprocess import CalledProcessError
+from typing import TYPE_CHECKING
+from typing import Literal
+
+import pbs_installer as pbi
+
+from poetry.core.constraints.version import Version
+
+from poetry.config.config import Config
+from poetry.console.exceptions import ConsoleMessage
+from poetry.console.exceptions import PoetryRuntimeError
+from poetry.utils.env.python import Python
+
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+
+BAD_PYTHON_INSTALL_INFO = [
+    "This could happen because you are missing platform dependencies required.",
+    "Please refer to https://gregoryszorc.com/docs/python-build-standalone/main/running.html#runtime-requirements "
+    "for more information about the necessary requirements.",
+    "Please remove the failing Python installation using <c1>poetry python remove <version></> before continuing.",
+]
+
+
+class PythonInstallerError(Exception):
+    pass
+
+
+class PythonDownloadNotFoundError(PythonInstallerError, ValueError):
+    pass
+
+
+class PythonInstallationError(PythonInstallerError, ValueError):
```

Comment:
Is there a reason to inherit from `PythonInstallerError` and `ValueError`?

Reply:
It was more for correctness as it is also an error due to a bad value while maintaining the base class for the package. Not strictly required.